contentcuration/contentcuration/frontend/shared/utils/validation.js

```diff
@@ -0,0 +1,397 @@
+  return [
+    value => Boolean(value) || ValidationErrors.MASTERY_MODEL_M_REQUIRED,
+    value => Number.isInteger(Number(value)) || ValidationErrors.MASTERY_MODEL_M_WHOLE_NUMBER,
+    value => value > 0 || ValidationErrors.MASTERY_MODEL_M_GT_ZERO,
+    value => value <= nValue || ValidationErrors.MASTERY_MODEL_M_LTE_N,
+  ];
+}
+
+export function getMasteryModelNValidators() {
+  return [
+    value => Boolean(value) || ValidationErrors.MASTERY_MODEL_N_REQUIRED,
+    value => Number.isInteger(Number(value)) || ValidationErrors.MASTERY_MODEL_N_WHOLE_NUMBER,
+    value => value > 0 || ValidationErrors.MASTERY_MODEL_N_GT_ZERO,
+  ];
+}
+
+// Node validation
+// These functions return an array of error codes
+export function validateNodeTitle(node) {
+  return getTitleValidators()
+    .map(validator => validator(node.title))
+    .filter(value => value !== true);
+}
+
+export function validateNodeLicense(node) {
+  const license = _getLicense(node);
+  return getLicenseValidators()
+    .map(validator => validator(license))
+    .filter(value => value !== true);
+}
+
+export function validateNodeCopyrightHolder(node) {
+  const license = _getLicense(node);
+  if (!license || !license.copyright_holder_required) {
+    return [];
+  }
+  return getCopyrightHolderValidators()
+    .map(validator => validator(node.copyright_holder))
+    .filter(value => value !== true);
+}
+
+export function validateNodeLicenseDescription(node) {
+  const license = _getLicense(node);
+  if (!license || !license.is_custom) {
+    return [];
+  }
+  return getLicenseDescriptionValidators()
+    .map(validator => validator(node.license_description))
+    .filter(value => value !== true);
+}
+
+export function validateNodeMasteryModel(node) {
+  const mastery = _getMasteryModel(node);
+  return getMasteryModelValidators()
+    .map(validator => validator(mastery))
+    .filter(value => value !== true);
+}
+
+export function validateNodeMasteryModelM(node) {
+  const mastery = _getMasteryModel(node);
+  if (!mastery || mastery.type !== MasteryModelsNames.M_OF_N) {
+    return [];
+  }
+  return getMasteryModelMValidators(mastery.n)
+    .map(validator => validator(mastery.m))
+    .filter(value => value !== true);
+}
+
+export function validateNodeMasteryModelN(node) {
+  const mastery = _getMasteryModel(node);
+  if (!mastery || mastery.type !== MasteryModelsNames.M_OF_N) {
+    return [];
+  }
+  return getMasteryModelNValidators()
+    .map(validator => validator(mastery.n))
+    .filter(value => value !== true);
+}
+
+/**
+ * @param  {Object} nodeDetails Node data like title, license, ...
+ * @param  {Array} assessmentItems An array of the node's assessment items
+ *                                 (for exercise nodes)
+ * @return {Boolean} Returns `true` if node details (and assessment items for exercises) are valid.
+ *                   Returns `false` otherwise.
+ */
+export function isNodeComplete(nodeDetails, assessmentItems = []) {
```

Comment:
Could you also add in a call to `validateNodeFiles`? We'll need this too as we don't want poeople to have video nodes with no files attached for instance